pageserver/src/pgdatadir_mapping.rs

```diff
@@ -919,8 +931,8 @@ impl<'a, T: DatadirTimeline> DatadirModification<'a, T> {
                 bail!("unexpected pending WAL record");
             }
         } else {
-            let last_lsn = self.tline.get_last_record_lsn();
-            self.tline.get(key, last_lsn)
+            let lsn = Lsn::max(self.tline.get_last_record_lsn(), self.last_flush_lsn);
+            self.tline.get(key, lsn)
```

Comment:
Does it work if we read at `self.lsn`? Seems simpler unless I'm missing something

Reply:
Yes, it should work. Prior to this PR, `self.lsn` was removed from the struct but added back in https://github.com/neondatabase/neon/pull/2094.

Reply:
updated in https://github.com/neondatabase/neon/pull/2172/commits/b90832dd981be98492f6779d9a1be6a0edc31694